### PR TITLE
Add join models for student resources

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds support for a student resources dropdown for scripts and unit groups, which will be shown in a dropdown similar to teacher resources.

Part of [PLAT-644](https://codedotorg.atlassian.net/browse/PLAT-644). The steps to this project will be:
1. (This PR) Add the models needed for student resources
2. (#39917, still in draft) Seed and serialize these models
3. Add an edit UI to the script and course edit pages. This will be a lot easier than the work for teacher resources as this is a new feature.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class CreateScriptsStudentResources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :scripts_student_resources do |t|
+      t.integer :script_id
+      t.integer :resource_id
+
+      t.index [:script_id, :resource_id], unique: true
+      t.index [:resource_id, :script_id]
```
Comment (Hamms) [inline dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb:8]: why do we need both these indexes?

Comment (davidsbailey) [reply dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb:8]: if you removed the second one, then querying by resource_id alone would be unindexed.

**Review by davidsbailey — APPROVED** [review]
new join tables with id columns sounds like the right approach to me.

File: dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb:8]: then why not just index on resource id?

Comment (davidsbailey) [reply dashboard/db/migrate/20210402163201_create_scripts_student_resources.rb:8]: good question! :-) seems like toss-up to me. I think this is just the pattern we've been following.

**Pull request merged**